src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -290,17 +290,17 @@ public static bool IsWindowsAzureManagedHosting(this IEnvironment environment)
         /// <returns><see cref="true"/> if running in Kubernetes environment; otherwise, false.</returns>
         public static bool IsAnyKubernetesEnvironment(this IEnvironment environment)
         {
-            return environment.IsKubernetesManagedHosting() || environment.IsManagedAppEnvironment();
+            return environment.IsKubernetesManagedHosting() || environment.IsContainerAppEnvironment();
         }
 
         /// <summary>
-        /// Gets a value indicating whether the application is running in Managed App environment.
+        /// Gets a value indicating whether the application is running in Container App environment.
         /// </summary>
         /// <param name="environment">The environment to verify.</param>
-        /// <returns><see cref="true"/> if running in Managed App environment; otherwise, false.</returns>
-        public static bool IsManagedAppEnvironment(this IEnvironment environment)
+        /// <returns><see cref="true"/> if running in Container App environment; otherwise, false.</returns>
+        public static bool IsContainerAppEnvironment(this IEnvironment environment)
         {
-            return !string.IsNullOrEmpty(environment.GetEnvironmentVariable(ManagedEnvironment));
+            return !string.IsNullOrEmpty(environment.GetEnvironmentVariable(ManagedEnvironment)) || !string.IsNullOrEmpty(environment.GetEnvironmentVariable(ConnectedEnvironment));
```

Comment:
So now we're saying that when running on ACA, either MANAGED_ENVIRONMENT or CONNECTED_ENVIRONMENT will be set. In what cases will one or the other be set? Kind of confusing, right?

Reply:
So basically ACA supports two modes. Managed Environment where customer container apps are running on Microsoft managed dataplane. The other is Connected Environment where customer container apps are running on customer managed ARC enabled kubernetes cluster which have ACA extension installed. 

Both Managed and Connected Environment are set from the dataplane. So at a time it can be either one of two. Apps cannot run on both at the same time.